API_Doc/credit_risk_POC_docwriter.py

```diff
@@ -0,0 +1,135 @@
+update_loan_operation = "UpdateLoan"
+update_loan_operation_method = "POST"
+update_loan_operation_expects = loan_class.id_
+update_loan_operation_returns = None
+update_loan_operation_returns_header = ["Content-Type", "Content-Length"]
+update_loan_operation_expects_header = []
+# status code
+update_loan_operation_status = [HydraStatus(code=200, desc="Loan class updated.")]
+
+add_loan_operation = "AddLoan"
+add_loan_operation_method = "PUT"
+add_loan_operation_expects = loan_class.id_
+add_loan_operation_returns = None
+add_loan_operation_returns_header = []
+add_loan_operation_expects_header = []
+# status code
+add_loan_operation_status = [HydraStatus(code=200, desc="Loan class updated.")]
+
+get_loan_operation = "GetLoan"
+get_loan_operation_method = "GET"
+get_loan_operation_expects = None
+get_loan_operation_returns = loan_class.id_
+get_loan_operation_returns_header = ["Content-Type", "Content-Length"]
+get_loan_operation_expects_header = []
+# Status code
+get_loan_operation_status = [HydraStatus(code=200, desc="Loan class returned.")]
+
+post = HydraClassOp(update_loan_operation,
+                    update_loan_operation_method,
+                    update_loan_operation_expects,
+                    update_loan_operation_returns,
+                    update_loan_operation_expects_header,
+                    update_loan_operation_returns_header,
+                    update_loan_operation_status)
+
+add = HydraClassOp(add_loan_operation,
+                   add_loan_operation_method,
+                   add_loan_operation_expects,
+                   add_loan_operation_returns,
+                   add_loan_operation_expects_header,
+                   add_loan_operation_returns_header,
+                   add_loan_operation_status)
+
+get = HydraClassOp(get_loan_operation,
+                   get_loan_operation_method,
+                   get_loan_operation_expects,
+                   get_loan_operation_returns,
+                   get_loan_operation_expects_header,
+                   get_loan_operation_returns_header,
+                   get_loan_operation_status)
+
+# adding property & operation to Loan classes
+loan_class.add_supported_prop(Total_balance_prop)
+loan_class.add_supported_prop(Origination_prop)
+loan_class.add_supported_op(post)
+loan_class.add_supported_op(get)
+loan_class.add_supported_op(add)
+borrower_class.add_supported_prop(LegalEntityIdentifier_prop)
+borrower_class.add_supported_prop(TotalAssets_prop)
+borrower_class.add_supported_prop(DateOfIncorporation_prop)
+
+api_doc.add_supported_class(loan_class)
+api_doc.add_supported_class(borrower_class)
+
+api_doc.add_baseResource()
+api_doc.add_baseCollection()
+# creating Entrypoint
+api_doc.gen_EntryPoint()
+# generating API_Doc
+doc = api_doc.generate()
+
+# saving the API_Doc
+if __name__ == "__main__":
+    """Print the complete sample Doc in api_doc_output.py."""
+    import json
+    dump = json.dumps(doc, indent=4, sort_keys=True)
```

Comment:
What you are doing here is not the proper way.
What you want to do is:
1. build the ApiDoc programmatically (ok done)
2. save the JSON-LD string in a file (`json.dumps(doc)` -> `ApiDoc.jsonld`)
3. if you want to have the file in a Python dictionary: create a new file `apidoc.py` that does: 
```
doc = None
with open("ApiDoc.jsonld") as d:
    doc = json.loads(d.read())
```
All this has to be put under `if __name__ == "__main__":`, it tells the Python interpreter to run all the code when the command `$ python creditrisk_poc_docwriter.py` is run; as it is now only the last lines will be run but `doc` will be undefined.

Reply:
Okay, will fix this.
I have taken the reference from the sample present in the hydrus itself.
https://github.com/HTTP-APIs/hydrus/blob/develop/hydrus/samples/doc_writer_sample.py

Reply:
that's totally wrong I agree. That is from 2017 though and conditions were quite different, for example Github did not have code reviews at the time... D)